### PR TITLE
Symbols

### PR DESCRIPTION
**Pull request by ralvapil**
image

Added a new Icon Variant called Symbols, refer to BDS0 [https://www.figma.com/file/f2S8WpnOhoT8GZpCViB5su/BDS0-%7C-Base?node-id=387%3A4522](url).

Work Done:
1. Updated Scaling in IconBase implementation, the scaling will now work with only 1 multiplier which will be the smaller of the width multiplier and the height multiplier. This will ensure that SVG will maintain its aspect ratio without overflowing on one axis. It will also ensure the scaling keeps the SVG within the "bounding box" aka the viewbox container we give it.
2. Added an IconBase variant called SymbolIcon to override the sizes used.
3. Added specialized components for each symbol, BlockSymbol, DoneSymbol, PendingSymbol, QASymbol, WarningSymbol
4. Allowed height and width to be passed down from the specialized component to the Icon Variants to allow icon specific overrides of height (this is only used for scaling and the icon will still be limited to the override sizes defined in SymbolIcon)
5. Renamed the color prop for icons to 'stroke' since I feel we should keep inline with css styling as much as possible.

These will be used within the InputText component and other components to come. There is something a little off about the WarningSymbol that I will talk to Sean about later.

### CODE REVIEW

File: icon/src/utils.ts [hunk 1]
```diff
@@ -1,4 +1,8 @@
-export const getScaleMultiplier = (selectedSize, width, height) => ({
-  widthMultiplier: selectedSize / width,
-  heightMultiplier: selectedSize / height,
+// use the minimum scale multiplier to prevent the svg from overflowing out of the view box
+export const getScaleMultiplier = (selectedSize, width, height) =>
+  Math.min(selectedSize / width, selectedSize / height);
```
Comment (deanlot) [inline icon/src/utils.ts:3]: can we write tests for the algo please?

File: icon/src/utils.ts [hunk 2]
```diff
@@ -1,4 +1,8 @@
...
+
+export const getCenteringTranslationValues = (size, width, height, scaleMultiplier) => ({
```
Comment (deanlot) [inline icon/src/utils.ts:5]: Can we declare a type here please.

File: icon/stories/Icon.stories.tsx [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
 
 export const IconSmall = () => <CheckIcon size="s" />;
 export const IconMedium = () => <CheckIcon />;
 export const IconLarge = () => <CheckIcon size="l" />;
 export const IconExtraLarge = () => <CheckIcon size="xl" />;
+
+export const SymbolSmall = () => <WarningSymbol size="s" />;
+export const SymbolMedium = () => <WarningSymbol size="m" />;
+export const SymbolLarge = () => <WarningSymbol size="l" />;
+export const SymbolExtraLarge = () => <WarningSymbol size="xl" />;
```
Comment (deanlot) [inline icon/stories/Icon.stories.tsx:15]: These might belong as their own documented component.

**Review by deanlot — APPROVED** [review]

File: icon/src/utils.ts (hunk 1 above)
Comment (ralvapil) [reply icon/src/utils.ts]: removed

**Review by jsaelhof — COMMENTED** [review]
Just one comment we can talk about and then this should be gtg.

**Pull request merged**